spec/requests/reviewables_controller_spec.rb

```diff
@@ -790,25 +790,94 @@ def perform_approve_phony(performed_by, args)
     describe "#destroy" do
       fab!(:user) { Fabricate(:user) }
 
-      before { sign_in(user) }
-
       it "returns 404 if the reviewable doesn't exist" do
+        sign_in(user)
         delete "/review/1234.json"
         expect(response.code).to eq("404")
       end
 
       it "returns 404 if the user can't see the reviewable" do
+        sign_in(user)
         queued_post = Fabricate(:reviewable_queued_post)
         delete "/review/#{queued_post.id}.json"
         expect(response.code).to eq("404")
       end
 
       it "returns 200 if the user can delete the reviewable" do
+        sign_in(user)
         queued_post = Fabricate(:reviewable_queued_post, created_by: user)
         delete "/review/#{queued_post.id}.json"
         expect(response.code).to eq("200")
         expect(queued_post.reload).to be_deleted
       end
+
+      it "denies attempts to destroy unowned reviewables" do
+        sign_in(admin)
+        queued_post = Fabricate(:reviewable_queued_post, created_by: user)
+        delete "/review/#{queued_post.id}.json"
+        expect(response.status).to eq(404)
+        # Reviewable is not deleted because request is not via API
+        expect(queued_post.reload).to be_present
+      end
```

Comment:
This tests proves that even though we now allow `@user.admin?` to satisfy `can_delete_reviewable_queued_post?` it will only work for admins when called via the API. This won't grant access to admins to go deleting reviewables for any user in the UI.